### PR TITLE
Store command output

### LINKED ISSUE #32: Add command output to MonitoredScheduledTask model
It would be nice to be able to see the scheduled command output in the `MonitoredScheduledTask` model table.
It should only be logged when the user calls `storeOutput()` on the scheduled task.

I can/will PR a solution for this. The issue is for completeness. 🙂

### PR DESCRIPTION
**Pull request by cretueusebiu**
Fixes #32 based on the [reply](https://github.com/spatie/laravel-schedule-monitor/issues/32#issuecomment-791436058)  by @Gummibeer

### CODE REVIEW

File: tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php [hunk 1]
```diff
@@ -153,4 +153,26 @@ public function it_works_correctly_when_a_task_is_run_in_the_background()
...
     }
+
+    /** @test */
+    public function it_stores_the_command_output()
+    {
+        TestKernel::replaceScheduledTasks(function (Schedule $schedule) {
+            $path = storage_path('logs/dummy-task.log');
+
+            $schedule
+                ->call(fn () => file_put_contents($path, 'dummy output'))
```
[the remark this answers is not in the document]
Comment (cretueusebiu) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: It doesn't work like that. When you define an anonymous function you're not in a new PHP process so you can't really output anything. But that `file_put_contents` simulates the output that a process would write.

[the remark this answers is not in the document]
Comment (Gummibeer) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: Simulating isn't a test for that. You can schedule a real command class for real output.

[the remark this answers is not in the document]
Comment (cretueusebiu) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: You can't really run real commands through the scheduler in tests, or at least not that easy. 
Laravel creates a new PHP process that runs your command separately of the test case.  And since  orchestra/testbench doesn't include the `artisan` script you can't run real commands through the scheduler in tests.

But if you look at the Laravel source code for the scheduler and how it processes the output, you'll see that basically the process that runs your command will pipe the output to a file inside the logs folder.

[the remark this answers is not in the document]
Comment (cretueusebiu) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: @freekmurze any thoughts on this ?

File: src/Models/MonitoredScheduledTask.php [hunk 2]
```diff
@@ -175,4 +176,26 @@ protected function createLogItem(string $type): MonitoredScheduledTaskLogItem
...
+
+        if ($event->task->output === $event->task->getDefaultOutput()) {
+            return null;
+        }
+
+        if (! is_file($event->task->output)) {
+            return null;
+        }
+
+        $output = file_get_contents($event->task->output);
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Models/MonitoredScheduledTask.php:197]: Thanks for explaining!

File: tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gummibeer) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: @cretueusebiu have you tested it in a fresh Laravel app with the inspire command or any other command with some output?
So far I see it to make the output logging work I have to manually call `storeOutput()` on the scheduled command? So I think that the README needs an update as well. 🤔

[the remark this answers is not in the document]
Comment (cretueusebiu) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: @Gummibeer  Yes, I've tested it by replacing the package with my fork and it works. Yes you actually have to call `storeOutput()` to store it.

@freekmurze  You mean when will `getEventTaskOutput` return something ? That's when `ScheduledTaskFinished` is dispatched and you called `storeOutput()`.

[the remark this answers is not in the document]
Comment (cretueusebiu) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: @freekmurze If you think this is more of an edge case or potentially bc, could we at least make the `MonitoredScheduledTask` model configurable ? That way the base model could be extended and just override the `markAsFinished` method with the new implementation.

[the remark this answers is not in the document]
Comment (cretueusebiu) [reply tests/ScheduledTaskSubscriber/ScheduledTaskSubscriberTest.php:164]: @Gummibeer opened #43 with an actual artisan command to test.
